File: app/src/main/java/com/selfmate/mes/selfmate/Models/ResultsDao.java
package com.selfmate.mes.selfmate.Models;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;


/**
 * Created by vicki_mes on 11/20/2017.
 */
@Dao
public interface ResultsDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertResults(Results results);

    @Update
    void updateResults(Results results);

    @Query("SELECT * FROM results WHERE uid=1")
    LiveData<Results> getResults();

}

File: app/src/main/java/com/selfmate/mes/selfmate/models/ResultsDao.kt
package com.selfmate.mes.selfmate.models

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*


/**
 * Created by vicki_mes on 11/20/2017.
 */
@Dao
interface ResultsDao {

    @Query("SELECT * FROM results WHERE uid=1")
    fun getResults(): LiveData<Results>

    @Query("SELECT * FROM results WHERE uid=1")
    fun getLiveResults(): Results


    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertResults(results: Results): Long

    @Update
    fun updateResults(results: Results)

}
